desktop/src/main.rs

```diff
@@ -508,3 +567,36 @@ fn run_timedemo(opt: Opt) -> Result<(), Box<dyn std::error::Error>> {
 
     Ok(())
 }
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // When linked with the windows subsystem windows won't automatically attach
+    // to the console of the parent process, so we do it explicitly. This fails
+    // silently if the parent has no console.
+    #[cfg(windows)]
+    unsafe {
+        use winapi::um::wincon::{AttachConsole, ATTACH_PARENT_PROCESS};
+        AttachConsole(ATTACH_PARENT_PROCESS);
+    }
+
+    env_logger::init();
+
+    let opt = Opt::parse();
+    let result = if opt.timedemo {
+        run_timedemo(opt)
+    } else {
+        let app = Box::leak(Box::new(App::new(opt)));
```

Comment:
I'm a little worried about this -- I understand the app is exiting, so who cares about the memory, but objects inside `App` might rely on their `Drop`s being run. One example I can think of offhand is the `StorageBackend`, which wants to `flush` on exit. I haven't tested, but I suspect with this change, save data will not explicitly save on exit in many games.

Reply:
I agree, but I think this is not handled even now. It's quite difficult because `event_loop.run` requires a `'static` lifetime for everything it references from within its callback.

Reply:
Sorry, had to remind myself how winit works! This is indeed fine (`App` gets moved into the event loop closure in `event_loop.run`, and this will get dropped when the event loop is destroyed), so ignore all of the above. But the Box isn't necessary; you can make `App::run(mut self)` so it consumes `self`.

This does make me think we should move the shutdown stuff into an explicit function, as I don't think `FreeConsole` at the end of `main` is getting called at all during normal execution.

Reply:
Thanks for the `App::run(mut self)` tip!
Good point regarding `FreeConsole`, but extracting an explicit shutdown function would require some more effort because it should also apply for `run_timedemo`. So maybe defer this for a future PR?

Reply:
Nevermind, I decided to do it in this PR.